doc/proposals/scorecard-user-tests.md

```diff
@@ -0,0 +1,154 @@
+}
+
+// JSONTestResult is a simplified version of the TestResult that only include the Name and Description of the Test field in TestResult
+type JSONTestResult struct {
+    // State is the final state of the test
+    State         State
+    // Name is the name of the test
+    Name          string
+    // Description describes what the test does
+    Description   string
+    // EarnedPoints is how many points the test received after running
+    EarnedPoints  int
+    // MaximumPoints is the maximum number of points possible for the test
+    MaximumPoints int
+    // Suggestions is a list of suggestions for the user to improve their score (if applicable)
+    Suggestions   []string
+    // Errors is a list of the errors that occured during the test (this can include both fatal and non-fatal errors)
+    Errors        []error
+}
+
+// State is a type used to indicate the result state of a Test.
+type State string
+
+const (
+    // UnsetState is the default state for a TestResult. It must be updated by UpdateState or by the Test.
+    UnsetState State = "unset"
+    // PassState occurs when a Test's ExpectedPoints == MaximumPoints.
+    PassState State = "pass"
+    // PartialPassState occurs when a Test's ExpectedPoints < MaximumPoints and ExpectedPoints > 0.
+    PartialPassState State = "partial_pass"
+    // FailState occurs when a Test's ExpectedPoints == 0.
+    FailState State = "fail"
+    // ErrorState occurs when a Test encounters a fatal error and the reported points should not be considered.
+    ErrorState State = "error"
+)
+```
+
+JSON output for `ScorecardResult` object (for the initial `v1alpha1` of the scorecard test objects):
+
+```json
+{
+    "error": 0,
+    "pass": 1,
+    "partial_pass": 1,
+    "fail": 0,
+    "total_tests": 2,
+    "total_score_percent": 71,
+    "tests": [
+        {
+            "state": "partial_pass",
+            "name": "Operator Actions Reflected In Status",
+            "description": "The operator updates the Custom Resources status when the application state is updated",
+            "earnedPoints": 2,
+            "maximumPoints": 3,
+            "suggestions": [
+                {
+                    "suggestion": "Operator should update status when scaling cluster down"
+                }
+            ],
+            "errors": []
+        },
+        {
+            "state": "pass",
+            "name": "Verify health of cluster",
+            "description": "The cluster created by the operator is working properly",
+            "earnedPoints": 1,
+            "maximumPoints": 1,
+            "suggestions": [],
+            "errors": []
+        }
+    ]
+}
+```
+
+This JSON output would make it simple for others to create scorecard plugins while keeping it simple for the scorecard
+to parse and integrate with the other tests. Each plugin would be considered a separate suite, and the full result of the scorecard
+would be a list of `ScorecardResult`s.
```

Comment:
Should the plugin output be the full `ScorecardTest` object so that the `TypeMeta` fields are included?

Reply:
Yes, the plugins should output a full ScorecardTest JSON. The scorecard can then make an array of the results from each plugin for the ScorecardResults section of the main scorecard output.